### PR TITLE
Fixes Bug Saic-67 : CF fails if there are no vms

### PR DESCRIPTION
**Pull request by oleksii-shyman**
- added if cases in 4 tasks

### CODE REVIEW

**Comment by Wasapl** [comment]
Merged build finished. Test PASSed.  http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/21/

**Comment by roman-verchikov** [comment]
awesome, +1

**Comment by miarmak** [comment]
Are you sure we need `if` statements here? Maybe it is a scenario problem and we should use here:
https://github.com/MirantisWorkloadMobility/CloudFerry/blob/devel/cloud/os2os.py#L179
the same as here:
https://github.com/MirantisWorkloadMobility/CloudFerry/blob/devel/cloud/os2os.py#L182
The same thing in the file scenario - just operate with the existing actions.
What do you think?

**Comment by oleksii-shyman** [comment]
I've just checked it - it doesn't work according to wrong scenario file parser. Cloudferry actually does not start at all (because we have infinite loop). I think even if can use Your approach - we still need that ifs, because it is proper way to check if key exists before operating with it

**Comment by miarmak** [comment]
Actually, it is a known issue about the scenario file parser. Known, but forgotten :) For now we should define **each** task in `scenario/tasks.yaml`, even for the same tasks. Have you tried it? Also, there shouldn't be such problem using `cloud/os2os.py` directly.

**Comment by oleksii-shyman** [comment]
@miarmak  -  This approach doesn't work even in os2os.py directly

**Comment by miarmak** [comment]
Yeah, that's true. Therefore it seems like problem exists in scheduler instead of file scenario parser as we thought before... 
But anyway, I think we can reuse this scheduler functionality and avoid these `if` statements with providing new task `is_not_vms`, that will be pretty similar to `is_end_iter` and make smth like this (f.e. in `os2os.py`):
`
is_not_instances = is_not_vms.IsNotVms(self.init)

        transport_instances_and_dependency_resources = \
            act_get_filter >> \
            act_get_info_inst >> \
            init_iteration_instance >> \
            act_check_needed_compute_resources >> \
            (is_not_instances | rename_info_iter) >> \
            get_next_instance >> \
            trans_one_inst >> \
            save_result >> \
            (is_instances | get_next_instance) >>\
            rename_info_iter >> \
            act_cleanup_images
        return transport_instances_and_dependency_resources
`

What do you think?

**Comment by roman-verchikov** [comment]
Is this pull request still valid?

**Comment by oleksii-shyman** [comment]
I think yes

**Comment by oleksii-shyman** [comment]
Max, it still doesn't work.
I think We have to keep that "if" solution

**Comment by Wasapl** [comment]
Merged build finished. Test PASSed.

**Comment by oleksii-shyman** [comment]
need review on that

**Comment by roman-verchikov** [comment]
Looks very neat, +1

File: scenario/tasks.yaml [hunk 1]
```diff
@@ -35,7 +35,9 @@ tasks:
    act_cleanup_images: ['CleanupImages']
    get_next_instance: ['GetInfoIter']
    rename_info_iter: ['RenameInfo', 'info_result', 'info']
+   rename_info_iter1: ['RenameInfo', 'info_result', 'info']
```
Comment (miarmak) [inline scenario/tasks.yaml]: Why do we need to create new instance of RenameInfo here?

File: scenario/migrate.yaml [hunk 2]
```diff
@@ -42,6 +42,7 @@ process:
           - init_iteration_instance_copy_var: True
           - init_iteration_instance_ref: True
       - check_needed_compute_resources: True
+      - check_instances: ['rename_info_iter1']
```
Comment (miarmak) [inline scenario/migrate.yaml]: It looks like in this case we will not skip all transport instances tasks and will go to the next task (`get_next_instance`) if there are no instances on the source. I guess we need to do the same as in the os2os, namely use the `rename_info_iter`.

**Comment by miarmak** [comment]
Overall looks good, but have a question about scenario case inline.

File: scenario/tasks.yaml (hunk 1 above)
Comment (oleksii-shyman) [inline scenario/tasks.yaml]: because scenario parser cannot parse 2 instances in the same file

Comment (miarmak) [inline scenario/tasks.yaml]: Why do we need to make 2 instances here? We can just go forward only with 1 `rename_info_iter` here.

**Comment by Wasapl** [comment]
Merged build finished. Test PASSed.

**Comment by miarmak** [comment]
Great! Thank you.
+1

**Comment by roman-verchikov** [comment]
+1

**Pull request merged**